backend/store/etcd/silenced_store.go

```diff
@@ -211,27 +211,22 @@ func (s *Store) arraySilencedEntries(ctx context.Context, resp *clientv3.GetResp
 		if err := unmarshal(kv.Value, silenced); err != nil {
 			return nil, &store.ErrDecode{Err: err}
 		}
+
+		var expire int64
 		leaseID := clientv3.LeaseID(kv.Lease)
 		if leaseID > 0 {
 			// legacy expiry mechanism
-			leaseID := clientv3.LeaseID(kv.Lease)
-			ttl, err := s.client.TimeToLive(ctx, leaseID)
-			if err != nil {
-				logger.WithError(err).Error("error setting TTL on silenced")
-				continue
-			}
-			silenced.Expire = ttl.TTL
```

Comment:
So, this code was intentionally left in so that when users upgrade to a newer version, their silences continue to work.

But perhaps it's fine if we take the plunge now.

What do you think?

Reply:
if we decide to drop support for the legacy silences (a breaking change, but a small one), then this whole logical branch should be removed.

Reply:
As discussed, I've re-used this code so we calculate instead the ExpireAt value!